src/tracer/MOM_tracer_registry.F90

```diff
@@ -434,16 +447,45 @@ subroutine register_tracer_diagnostics(Reg, h, Time, diag, G, GV, US, use_ALE, u
     if (Tr%id_hbd_dfx_2d > 0) call safe_alloc_ptr(Tr%hbd_dfx_2d,IsdB,IedB,jsd,jed)
     if (Tr%id_hbd_dfy_2d > 0) call safe_alloc_ptr(Tr%hbd_dfy_2d,isd,ied,JsdB,JedB)
 
+    ! The following diagnostics for generic tracer budgets were
+    ! originally developed by Enhui Lao, Fan Yang, and Mathieu Poupon.
+
     Tr%id_adv_xy = register_diag_field('ocean_model', trim(shortnm)//"_advection_xy", &
         diag%axesTL, Time, &
-        'Horizontal convergence of residual mean advective fluxes of '//trim(lowercase(flux_longname)), &
-        conv_units, v_extensive=.true., conversion=Tr%conv_scale*US%s_to_T)
+        'Horizontal convergence of residual mean advective fluxes of '//&
+        trim(lowercase(flux_longname)), conv_units, v_extensive=.true., &
+        conversion=Tr%conv_scale*US%s_to_T)
+    Tr%id_difc_xy = register_diag_field('ocean_model',trim(shortnm)//"_diffusionc_xy", &
+        diag%axesTL, Time, "Horizontal convergence of residual mean diffusive fluxes of "//trim(shortnm), &
```

Comment:
@andrew-c-ross, I thought `diffusionc_xy` was for concentration and `diffusion_xy` was for content? Do you think it would be good to update the descriptions here, as well as on line 462, to reflect the difference?

Reply:
Thanks, I forgot to check the units and descriptions for the diagnostics on the MOM6 side. I think it should be okay now---I added "concentration" to the description of the c diagnostics and made it so the diagnostics for _diffusion_xy come out as mol/kg m s-1, the same as _advection_xy.

Reply:
For example:
```
"sio4_diffusionc_xy"  [Unused]
    ! modules: {ocean_model,ocean_model_z,ocean_model_rho2,ocean_model_d2,ocean_model_z_d2,ocean_model_rho2_d2}
    ! long_name: Horizontal convergence of residual mean diffusive fluxes of sio4 concentration
    ! units: mol/kg s-1
    ! cell_methods: xh:mean yh:mean zl:mean area:mean
    ! variants: {sio4_diffusionc_xy,sio4_diffusionc_xy_xyave}
"sio4_diffusion_xy"  [Unused]
    ! modules: {ocean_model,ocean_model_z,ocean_model_rho2,ocean_model_d2,ocean_model_z_d2,ocean_model_rho2_d2}
    ! long_name: Horizontal convergence of residual mean diffusive fluxes of sio4
    ! units: mol/kg m s-1
    ! cell_methods: xh:mean yh:mean zl:mean area:mean
    ! variants: {sio4_diffusion_xy,sio4_diffusion_xy_xyave}
```